django/db/backends/oracle/base.py

```diff
@@ -238,6 +240,38 @@ def __init__(self, *args, **kwargs):
         )
         self.features.can_return_columns_from_insert = use_returning_into
 
+    @property
+    def is_pool(self):
+        return self.settings_dict["OPTIONS"].get("pool")
+
+    @property
+    def pool(self):
+        if not self.settings_dict["OPTIONS"].get("pool"):
+            return None
+
+        # Verify that we are not running with persistent connections
+        if self.settings_dict.get("CONN_MAX_AGE", 0) != 0:
+            raise ImproperlyConfigured(
+                """Pooling doesn't support persistent connections, unset
+                   conn_max_age"""
+            )
+        # Pooling feature is only supported for oracledb
+        if not is_oracledb:
+            raise ImproperlyConfigured("Not supported by cx_Oracle install oracledb")
+
+        if self.alias not in self._connection_pools:
+            connect_kwargs = self.get_connection_params()
+            del connect_kwargs["pool"]
+            pool = Database.create_pool(
+                user=self.settings_dict["USER"],
+                password=self.settings_dict["PASSWORD"],
+                dsn=dsn(self.settings_dict),
+                **connect_kwargs,
+            )
+            self._connection_pools.setdefault(self.alias, pool)
+
+        return self._connection_pools[self.alias]
```

Comment:
New functionality should be tested, you can check new tests in #17914 for an inspiration.